Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp

```diff
@@ -857,6 +864,16 @@ void wpeToplevelWaylandViewVisibilityChanged(WPEToplevelWayland* toplevel, WPEVi
     }
 }
 
+gboolean wpeToplevelWaylandHasSyncObject(WPEToplevelWayland* toplevel)
+{
+    return !!toplevel->priv->sync;
+}
+
+struct zwp_linux_surface_synchronization_v1* wpeToplevelWaylandGetSyncListener(WPEToplevelWayland* toplevel)
+{
+    return toplevel->priv->sync;
+}
```

Comment:
These could be just one, a getter of the sync object that can also be used to check if it's present.